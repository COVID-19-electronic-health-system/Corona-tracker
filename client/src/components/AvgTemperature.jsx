import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { makeStyles } from '@material-ui/core/styles';
import { Grid, Typography } from '@material-ui/core';

const useStyles = makeStyles({
  root: {
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    justifyContent: 'flex-start',
  },
});

const AvgTemperature = props => {
  const { observations } = props;

  const classes = useStyles();

  const averageTemperature = observations.length
    ? Math.round(
        observations.reduce((sum, record) => {
          return sum + record.physical.feverSeverity * 10;
        }, 0) / observations.length
      ) / 10
    : '--';

  return (
    <Grid className={classes.root}>
      <Typography>Average Temperature:</Typography>
      <Typography>{averageTemperature}&#8457;</Typography>
    </Grid>
  );
};

AvgTemperature.propTypes = {
  observations: PropTypes.arrayOf(Object).isRequired,
};

const mapStateToProps = state => {
  return {
    observations: state.observationsReducer.observations,
  };
};

export default connect(mapStateToProps)(AvgTemperature);
