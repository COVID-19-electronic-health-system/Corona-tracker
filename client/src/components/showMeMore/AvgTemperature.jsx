import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { makeStyles } from '@material-ui/core/styles';
import { Paper, Typography } from '@material-ui/core';

const useStyles = makeStyles(theme => ({
  root: {
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    justifyContent: 'flex-start',
    maxWidth: '200px',
    backgroundColor: '#bbcef9',
    padding: theme.spacing(1),
  },
}));

const AvgTemperature = props => {
  const { observations, tempUnit } = props;

  const classes = useStyles();

  const averageTemperature = observations.length
    ? Math.round(
        observations.reduce((sum, record) => {
          return sum + record.physical.feverSeverity * 10;
        }, 0) / observations.length
      ) / 10
    : '--';

  return (
    <Paper className={classes.root}>
      <Typography variant="overline">Average Temperature:</Typography>
      {tempUnit === 'fahrenheit' ? (
        <Typography variant="body2">{averageTemperature}&#8457;</Typography>
      ) : (
        <Typography variant="body2">{averageTemperature}&#8451;</Typography>
      )}
    </Paper>
  );
};

AvgTemperature.propTypes = {
  observations: PropTypes.arrayOf(Object).isRequired,
  tempUnit: PropTypes.string.isRequired,
};

const mapStateToProps = state => {
  return {
    observations: state.observationsReducer.observations,
    tempUnit: state.onboardingReducer.tempUnit,
  };
};

export default connect(mapStateToProps)(AvgTemperature);
