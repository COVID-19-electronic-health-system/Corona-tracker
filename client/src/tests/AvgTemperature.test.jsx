import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import AvgTemperature from 'components/showMeMore/AvgTemperature';

const mockStore = configureMockStore([thunk]);
const mockObservationsReducerState = {
  observations: [
    {
      date: 1586830448900,
      physical: { feverSeverity: 98.6 },
    },
    {
      date: 1586910193682,
      physical: { feverSeverity: 100.3 },
    },
    {
      date: 1586966651099,
      physical: { feverSeverity: 99 },
    },
  ],
};

describe('Average Temperature', () => {
  it('renders AvgTemperature component', () => {
    const store = mockStore({
      observationsReducer: mockObservationsReducerState,
    });

    const wrapper = shallow(
      <Provider store={store}>
        <AvgTemperature />
      </Provider>
    );
    expect(wrapper.find('AvgTemperature')).toMatchSnapshot();
  });
});
