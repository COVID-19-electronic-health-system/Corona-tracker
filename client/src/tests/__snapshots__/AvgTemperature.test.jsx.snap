// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Average Temperature renders AvgTemperature component 1`] = `ShallowWrapper {}`;
